scanner/updater/jsonblob/jsonblob.go

```diff
@@ -0,0 +1,152 @@
+// Package jsonblob is a fork of ClairCore's jsonblob with minimal changes to
+// support individual record iteration.
+//
+// TODO(ROX-24333): Remove this package once ClairCore supports it, or become
+//                  obsolete by other formats.
+
+package jsonblob
+
+import (
+	"bufio"
+	"encoding/json"
+	"io"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/pkg/errors"
+	"github.com/quay/claircore"
+	"github.com/quay/claircore/libvuln/driver"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+type iter2[X, Y any] func(yield func(X, Y) bool)
+
+// RecordIter iterates over records of an update operation.
+type RecordIter iter2[*claircore.Vulnerability, *driver.EnrichmentRecord]
+
+// OperationIter iterates over operations, offering a nested iterator for records.
+type OperationIter iter2[*driver.UpdateOperation, RecordIter]
+
+// Iterate iterates over each record serialized in the [io.Reader] grouping by
+// update operations. It returns an OperationIter, which is an iterator over each
+// update operation with a nested iterator for the associated vulnerability
+// entries, and an error function, to check for iteration errors.
+func Iterate(r io.Reader) (OperationIter, func() error) {
+	var err error
+	var de diskEntry
+
+	d := json.NewDecoder(r)
+	err = d.Decode(&de)
```

Comment:
To keep the reference to the decoder the same as the iter() is called repeatedly.

Reply:
Sorry I don't fully understand this ^. Can you rephrase?

I think we'd have the same thing if we did the following instead:

```
it := func(yield func(*driver.UpdateOperation, RecordIter) bool) {
        var de diskEntry
        err = d.Decode(&de)
        for err == nil {
                ...
        }
        ...
}
```

This defers the first read to the first call to the iterator instead of upon initialization. Honestly, it's not a big deal, and I'm ok with either way, but I was just curious if that was considered

Reply:
To maintain consistency, we want to keep the reference to the decoder unchanged each time `iter()` is called. That applies to `d.Decode(&de)`. If not, we would try to read from the reader again if the caller calls `it()` more than once after `errF() != nil`.

> This defers the first read to the first call to the iterator instead of upon initialization.

Yeah, I know, it doesn't make much of a difference because two `it()`s is not part of the contract. But it is cleaner and safer this way.

Reply:
Any other questions?